go/ops.go

```diff
@@ -123,6 +129,16 @@ func (op *LeafOp) CheckAgainstSpec(spec *ProofSpec) error {
 
 // CheckAgainstSpec will verify the InnerOp is in the format defined in spec
 func (op *InnerOp) CheckAgainstSpec(spec *ProofSpec, b int) error {
+	if spec == nil || op == nil {
+		return errors.New("op and spec must be both non-nil")
+	}
+	if spec.InnerSpec == nil {
+		return errors.New("spec.InnerSpec must be non-nil")
+	}
+	if spec.LeafSpec == nil {
+		return errors.New("spec.LeafSpec must be non-nil")
+	}
```

Comment:
I might be missing something, but when I commented out this code, no test fails. Is the fuzzer correctly generating a test case for this scenario? The fuzzing test fails if I remove the checks for `InnerSpec` and `InnerSpec.ChildSize`.

Reply:
The fuzzer def flagged this condition too in mutations. In any case though, it is a defensive check, I highly recommend keeping that.